### PR TITLE
yosys-uhdm: use uhdm-plugin; add gperftools and libunwind packages

### PR DESCRIPTION
**Pull request by kamilrakoczy**
This PR replaces building uhdm as a part of yosys frontend to external plugin.

### CODE REVIEW

**Comment by kamilrakoczy** [comment]
It was necessary to add building ``libunwind`` and ``gperftools`` packages to get rid if ``conda-forge`` channel. ``conda-forge`` channel was creating conflicts when we don't want to use it, but we still want to install ``yosys-uhdm`` package (e.g. ``fpga-tool-perf``).

**Review by ajelinski — CHANGES_REQUESTED** [review]
The most important change is adding `run_exports` to `libunwind`.

BTW. I know some of the `yosys-uhdm` changes are outside the scope of this PR but let's fix it. Some of these only make the recipe unnecessarily complicated so it'll make the recipe easier to modify and maintain.

File: misc/gperftools/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  number: {{ environ.get('DATE_NUM') }}
+  string: {{ environ.get('DATE_STR') }}
+  script_env:
+    - CI
+
+requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ pin_compatible('libunwind', max_pin='x.x') }}  # [linux]
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply misc/gperftools/meta.yaml]: Fixed

File: misc/libunwind/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
+# Use `conda-build-prepare` before building for a better version string.
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply misc/libunwind/meta.yaml]: Removed

File: misc/libunwind/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+
+package:
+  name: libunwind
+  version: {{ version }}
+
+source:
+  url: http://download.savannah.nongnu.org/releases/libunwind/libunwind-{{ version }}.tar.gz
+  sha256: 90337653d92d4a13de590781371c604f9031cdb50520366aa1e3a91e1efb1017
+  patches:
+    - lib64.patch
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply misc/libunwind/meta.yaml]: Ok, removed

File: misc/libunwind/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+build:
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply misc/libunwind/meta.yaml:11]: Fixed

File: syn/yosys-uhdm/build.sh [hunk 5]
```diff
@@ -3,17 +3,27 @@
...
 export CXXFLAGS="$CXXFLAGS -I$BUILD_PREFIX/include"
 export LDFLAGS="$CXXFLAGS -L$BUILD_PREFIX/lib -lrt -ltinfo"
 export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$BUILD_PREFIX/lib"
 
-cd Surelog && make PREFIX=$PWD/../image release install -j$CPU_COUNT && cd ..
+cd Surelog && \
+	cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH -DCMAKE_POSITION_INDEPENDENT_CODE=ON -S . -B build && \
+	cmake --build build -j$(nproc) && \
+	cmake --install build && \
+	cd ..
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh:12]: Ok, I left it as it is for now.

File: syn/yosys-uhdm/meta.yaml [hunk 6]
```diff
@@ -42,8 +45,11 @@ requirements:
     - ncurses
     - readline
     - libffi
+    - gperftools
+    - libunwind
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/meta.yaml]: Done

File: syn/yosys-uhdm/build.sh [hunk 7]
```diff
@@ -3,17 +3,27 @@
 set -e
 set -x
 
+INSTALL_PATH=${PWD}/image
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: True, I removed ``INSTALL_PATH``

File: syn/yosys-uhdm/build.sh [hunk 8]
```diff
@@ -3,17 +3,27 @@
...
+	cmake -DCMAKE_BUILD_TYPE=Release -DCMAKE_INSTALL_PREFIX=$INSTALL_PATH -DCMAKE_POSITION_INDEPENDENT_CODE=ON -S . -B build && \
+	cmake --build build -j$(nproc) && \
+	cmake --install build && \
+	cd ..
 
 #Create aliases for gcc/gxx as `abc` uses them directly in Makefile
 alias gcc=x86_64-conda_cos6-linux-gnu-gcc
 alias gxx=x86_64-conda_cos6-linux-gnu-gcc
-make ENABLE_READLINE=0 CONFIG=conda-linux PROGRAM_PREFIX=uhdm- PREFIX=$PWD/image install -j$CPU_COUNT
+cd yosys && make ENABLE_READLINE=0 CONFIG=conda-linux PROGRAM_PREFIX=uhdm- PREFIX=$INSTALL_PATH install -j$(nproc) && cd ..
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: Fixed

File: syn/yosys-uhdm/build.sh [hunk 9]
```diff
@@ -3,17 +3,27 @@
...
+
+export PATH=$INSTALL_PATH/bin:${PATH}
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: Done

File: syn/yosys-uhdm/build.sh [hunk 10]
```diff
@@ -3,17 +3,27 @@
...
+
+UHDM_INSTALL_DIR=$INSTALL_PATH make -C $PWD/yosys-symbiflow-plugins/ install -j$(nproc)
 
 mkdir -p "$PREFIX/"
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: Done

File: syn/yosys-uhdm/build.sh [hunk 11]
```diff
@@ -3,17 +3,27 @@
...
 
 #Create aliases for gcc/gxx as `abc` uses them directly in Makefile
 alias gcc=x86_64-conda_cos6-linux-gnu-gcc
 alias gxx=x86_64-conda_cos6-linux-gnu-gcc
-make ENABLE_READLINE=0 CONFIG=conda-linux PROGRAM_PREFIX=uhdm- PREFIX=$PWD/image install -j$CPU_COUNT
+cd yosys && make ENABLE_READLINE=0 CONFIG=conda-linux PROGRAM_PREFIX=uhdm- PREFIX=$INSTALL_PATH install -j$(nproc) && cd ..
+
+export PATH=$INSTALL_PATH/bin:${PATH}
+
+UHDM_INSTALL_DIR=$INSTALL_PATH make -C $PWD/yosys-symbiflow-plugins/ install -j$(nproc)
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: Fixed

File: syn/yosys-uhdm/meta.yaml [hunk 12]
```diff
@@ -30,6 +31,8 @@ requirements:
     - flex
     - libuuid
     - pkg-config
+    - gperftools
+    - libunwind
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/meta.yaml]: Fixed

File: syn/yosys-uhdm/build.sh [hunk 13]
```diff
@@ -3,17 +3,27 @@
 set -e
 set -x
 
+INSTALL_PATH=${PWD}/image
+
 export PKG_CONFIG_PATH="$BUILD_PREFIX/lib/pkgconfig/"
 export CXXFLAGS="$CXXFLAGS -I$BUILD_PREFIX/include"
 export LDFLAGS="$CXXFLAGS -L$BUILD_PREFIX/lib -lrt -ltinfo"
 export LD_LIBRARY_PATH="$LD_LIBRARY_PATH:$BUILD_PREFIX/lib"
```
[the remark this answers is not in the document]
Comment (kamilrakoczy) [reply syn/yosys-uhdm/build.sh]: Done

**Review by ajelinski — APPROVED** [review]

**Pull request merged**